src/taipy/gui_core/_context.py

```diff
@@ -271,11 +298,110 @@ def submit_entity(self, state: State, id: str, action: str, payload: t.Dict[str,
         entity = core_get(entity_id)
         if entity:
             try:
-                core_submit(entity)
+                jobs = core_submit(entity)
+                if action := data.get("on_submission_change"):
+                    if callable(self.gui._get_user_function(action)):
+                        job_ids = [j.id for j in (jobs if isinstance(jobs, list) else [jobs])]
+                        client_id = self.gui._get_client_id()
+                        module_context = self.gui._get_locals_context()
+                        sub_id = jobs[0].submit_id if isinstance(jobs, list) else jobs.submit_id
+                        with self.submissions_lock:
+                            self.client_jobs_by_submission[sub_id] = (
+                                client_id,
+                                module_context,
+                                action,
+                                entity_id,
+                                _SubmissionStatus.SUBMITTED,
+                                job_ids,
+                            )
+                        self.scenario_status_callback(jobs[0].id if isinstance(jobs, list) else jobs.id)
+                    else:
+                        _warn(f"on_submission_change(): '{action}' is not a valid function.")
                 state.assign(_GuiCoreContext._SCENARIO_VIZ_ERROR_VAR, "")
             except Exception as e:
                 state.assign(_GuiCoreContext._SCENARIO_VIZ_ERROR_VAR, f"Error submitting entity. {e}")
 
+    def _get_submittable_status(self, jobs_ids: t.List[str]):
+        status = _SubmissionStatus.UNDEFINED
+        blocked = False
+        waiting = False
+        running = False
+        completed = False
+        for id in jobs_ids:
+            job = core_get(id)
```

Comment:
I'm a bit skeptic about the Core side performance of this... get() for each job...

Reply:
maybe @jrobinAV can chip in